lib/sles4sap/sap_deployment_automation_framework/configure_tfvars.pm

```diff
@@ -116,34 +130,126 @@ sub set_workload_vnet_name {
     set_var('SDAF_SUT_VNET_NAME', 'OpenQA-' . $args{job_id});
 }
 
-=head2 set_vm_image_parameters
-
-    set_vm_db_image_parameters([job_id=>'123456']);
+=head2 set_image_parameters
 
-=over
-
-=item * B<$job_id>: Specify job id to be used. Default: current deployment job ID
-
-=back
+    set_image_parameters();
 
 Sets OpenQA parameters required for replacing tfvars template variables for database VM image.
 
 =cut
 
-sub set_db_image_parameters {
+sub set_image_parameters {
     my %params;
     # Parse image ID supplied by OpenQA parameter 'PUBLIC_CLOUD_IMAGE_ID'
-    my @variable_names = qw(SDAF_DB_IMAGE_PUBLISHER SDAF_DB_IMAGE_OFFER SDAF_DB_IMAGE_SKU SDAF_DB_IMAGE_VERSION);
+    my @variable_names = qw(SDAF_IMAGE_PUBLISHER SDAF_IMAGE_OFFER SDAF_IMAGE_SKU SDAF_IMAGE_VERSION);
     # This maps a variable name from array @variable names to value from delimited 'PUBLIC_CLOUD_IMAGE_ID' parameter
     # Order is important here
     @params{@variable_names} = split(':', get_required_var('PUBLIC_CLOUD_IMAGE_ID'));
 
     # Add all remaining parameters with static values
-    $params{SDAF_DB_IMAGE_OS_TYPE} = 'LINUX';    # this can be modified in case of non linux images
-    $params{SDAF_DB_SOURCE_IMAGE_ID} = '';    # for supplying uploaded image - not implemented yet
-    $params{SDAF_DB_IMAGE_TYPE} = 'marketplace';
+    $params{SDAF_IMAGE_OS_TYPE} = 'LINUX';    # this can be modified in case of non linux images
+    $params{SDAF_SOURCE_IMAGE_ID} = '';    # for supplying uploaded image - not implemented yet
+    $params{SDAF_IMAGE_TYPE} = 'marketplace';
 
     foreach (keys(%params)) {
         set_var($_, $params{$_});
     }
 }
+
+=head2 set_hana_db_parameters
+
+    set_hana_db_parameters(components=>['db_install', 'db_ha']);
+
+Sets tfvars Database HA parameters according to scenario defined by B<$args{components}>.
+
+=over
+
+=item * B<components>: B<ARRAYREF> of components that should to be installed. Check function B<validate_components> for available options.
+
+=back
+
+=cut
+
+sub set_hana_db_parameters {
+    my (%args) = @_;
+    # Enable HA cluster
+    int(grep /ha/, @{$args{components}})
```

Comment:
This `int()` is not really needed, but the brackets are needed. Suggestion:
```
(grep /ha/, @{$args{components}}) ? set_var('SDAF_HANA_HA_SETUP', 'true') : set_var('SDAF_HANA_HA_SETUP', 'false');
```
Or better yet:
```
set_var('SDAF_HANA_HA_SETUP', (grep /ha/, @{$args{components}}) ? 'true' : 'false');
```
This way line is shorter, you avoid typing `SDAF_HANA_HA_SETUP` twice, etc.

Reply:
Thanks. That last version is cool. I am changing it to that one.